api/app/utils.py

```diff
@@ -10,9 +11,20 @@ def create_hash():
 
 def get_version_info() -> dict:
     """Reads the version info baked into src folder of the docker container"""
-    version_json = {
+    release_please_manifest_location = "./.versions.json"
+    manifest_versions = None
+    version_json = {}
+    image_tag = "unknown"
+
+    manifest_versions = json.loads(_get_file_contents(release_please_manifest_location))
```

Comment:
This is going to throw a JSONDecodeError if the file doesn't exist: 

```
>>> json.loads("unknown")
Traceback (most recent call last):
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/pydevconsole.py", line 364, in runcode
    coro = func()
           ^^^^^^
  File "<input>", line 1, in <module>
  File "/Users/matthewelwell/.asdf/installs/python/3.11.4/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthewelwell/.asdf/installs/python/3.11.4/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthewelwell/.asdf/installs/python/3.11.4/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

This highlights that we don't have a test which exercises this path!

Reply:
This is perhaps a bit neater:

```python
UNKNOWN = "unknown"
VERSIONS_INFO_FILE_LOCATION = "./.versions.json"

...

def get_version_info() -> dict:
    """Reads the version info baked into src folder of the docker container"""
    version_json: dict = {}
    image_tag: str = UNKNOWN

    versions_file_content: str = _get_file_contents(VERSIONS_INFO_FILE_LOCATION)

    if versions_file_content:
        versions_data: dict = json.loads(versions_file_content.replace("\n", ""))
        version_json["package_versions"] = versions_data
        image_tag = versions_data["."]

    return version_json | {
        "ci_commit_sha": _get_file_contents("./CI_COMMIT_SHA") or UNKNOWN,
        "image_tag": image_tag,
        "is_enterprise": pathlib.Path("./ENTERPRISE_VERSION").exists(),
    }


def _get_file_contents(file_path: str) -> typing.Optional[str]:
    """Attempts to read a file from the filesystem and return the contents"""
    try:
        with open(file_path) as f:
            return f.read()
    except FileNotFoundError:
        return None
```

Reply:
OK refactored

Reply:
Well, you've _half_ refactored. I'd rather that the `_get_file_contents` function was so logic-aware, but whatever.

Reply:
I would really like to see a test that exercises this code if the file doesn't exist though!

Reply:
OK test added